machine_translation/prepare_data.py

```diff
@@ -160,7 +161,7 @@ def merge_parallel(src_filename, trg_filename, merged_filename):
                 while True:
                     lline = left.readline()
                     rline = right.readline()
-                    if (lline == '') or (rline == ''):
+                    if (lline == '') and (rline == ''):
```

Comment:
`eof` is specified by empty string, when it is `and` meaning parallel text reached the end, `or` was breaking whenever it is just a regular empty line met in either of the text files, resulting smaller shuffled parallel text.